lnwallet/channel.go

```diff
@@ -291,6 +320,100 @@ type PaymentDescriptor struct {
 	isForwarded bool
 }
 
+// PayDescsFromRemoteLogUpdates converts a slice of LogUpdates received by a
+// the remote peer into PaymentDescriptors to inform a link's forwarding
+// decisions.
+func (lc *LightningChannel) PayDescsFromRemoteLogUpdates(height uint64,
+	logUpdates []channeldb.LogUpdate) []*PaymentDescriptor {
+
+	lc.Lock()
+	defer lc.Unlock()
+
+	var payDescs []*PaymentDescriptor
+	for i, logUpdate := range logUpdates {
+		var pd *PaymentDescriptor
+		switch wireMsg := logUpdate.UpdateMsg.(type) {
+
+		// For offered HTLC's, we'll map that to a PaymentDescriptor
+		// with the type Add, ensuring we restore the necessary fields.
+		// From the PoV of the commitment chain, this HTLC was included
+		// in the remote chain, but not the local chain.
+		case *lnwire.UpdateAddHTLC:
+			// First, we'll map all the relevant fields in the
+			// UpdateAddHTLC message to their corresponding fields in the
+			// PaymentDescriptor struct. We also set addCommitHeightRemote
+			// as we've included this HTLC in our local commitment chain
+			// for the remote party.
+			pd = &PaymentDescriptor{
+				RHash:     wireMsg.PaymentHash,
+				Timeout:   wireMsg.Expiry,
+				Amount:    wireMsg.Amount,
+				EntryType: Add,
+				HtlcIndex: wireMsg.ID,
+				LogIndex:  logUpdate.LogIndex,
+				SourceRef: &channeldb.AddRef{
+					Height: height,
+					Index:  uint16(i),
+				},
+			}
+			pd.OnionBlob = make([]byte, len(wireMsg.OnionBlob))
+			copy(pd.OnionBlob[:], wireMsg.OnionBlob[:])
+
+		// For HTLC's we we're offered we'll fetch the original offered HTLc
+		// from the remote party's update log so we can retrieve the same
```

Comment:
Similar comment here about the comment not matching the new code it was transplanted to. Applies to the other comments on the switch cases in this function as well.

Reply:
Fixed.